proto/src/main/proto/tech/figure/objectstore/gateway/gateway.proto

```diff
@@ -26,12 +28,26 @@ message FetchObjectResponse {
 message PutObjectRequest {
   ObjectWithMeta object = 1;
   repeated io.provenance.scope.PublicKey additional_audience_keys = 2;
+  bool use_requester_key = 3;
 }
 
 message PutObjectResponse {
   string hash = 1; // base64-encoded object hash (sha-256 of provided ObjectWithMeta [object] serialized proto)
 }
 
+/**
+ * Allows registering pre-existing object-store objects so that they may be fetched and managed in the same way as objects troed via
+ * the PutObject route
+ */
+message RegisterExistingObjectRequest {
+  string hash = 1; // the hash of the object in object store (the caller's access to this object will be verified during the course of this call)
+  repeated string grantee_address = 2; // addresses that should have access to fetch this object via the FetchObjectByHash rpc
+}
+
+message RegisterExistingObjectResponse {
+
```

Comment:
wasn't sure if anything made sense in these responses...

Reply:
Could make sense to mirror the input, just in case someone is doing some large async batch of these and is doing verification upon completion with record tracking or something

Reply:
good idea, I just added/set a field called 'request' in these